openquake/hazardlib/geo/surface/base.py

```diff
@@ -427,6 +427,40 @@ def get_middle_point(self):
 
         return mesh.get_middle_point()
 
+    def get_resampled_top_edge(self, angle_var=0.1):
+        """
+        Compute the top edge of the rupture plane.
```

Comment:
This methods computes a simplified representation of a fault top edge by removing the points that are not describing a change of direction, provided a certain tolerance angle.